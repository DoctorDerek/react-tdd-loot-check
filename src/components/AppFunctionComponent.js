import React from "react"

const AppFunctionComponent = () => {
  return (
    <section>
      <h2>Loot Check (Function Component)</h2>
    </section>
  )
}

export default AppFunctionComponent
