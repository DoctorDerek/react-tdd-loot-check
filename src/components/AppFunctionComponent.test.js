import React from "react"
import { mount } from "enzyme"
import AppFunctionComponent from "./AppFunctionComponent"

let app = mount(<AppFunctionComponent />)

describe("App (Function Component)", () => {
  beforeEach(() => {
    // Clean up between tests:
    app = mount(<AppFunctionComponent />)
  })

  // It Renders.
  it("Renders correctly", () => {
    expect(app).toMatchSnapshot()
  })
})
