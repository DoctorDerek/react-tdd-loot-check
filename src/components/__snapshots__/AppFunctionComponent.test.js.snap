// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App (Function Component) Renders correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
    "wrapInvoke": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": <section>
        <h2>
                Loot Check (Function Component)
        </h2>
</section>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Loot Check (Function Component)
</h2>,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <h2>
            Loot Check (Function Component)
</h2>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loot Check (Function Component)",
          },
          "ref": null,
          "rendered": Array [
            "Loot Check (Function Component)",
          ],
          "type": "h2",
        },
      ],
      "type": "section",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": <section>
          <h2>
                    Loot Check (Function Component)
          </h2>
</section>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            Loot Check (Function Component)
</h2>,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <h2>
              Loot Check (Function Component)
</h2>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loot Check (Function Component)",
            },
            "ref": null,
            "rendered": Array [
              "Loot Check (Function Component)",
            ],
            "type": "h2",
          },
        ],
        "type": "section",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
  Symbol(enzyme.__linkedRoots__): Array [],
  Symbol(enzyme.__unrendered__): <AppFunctionComponent />,
  Symbol(enzyme.__updatedBy__): null,
}
`;
